code/modules/mob/living/simple_animal/shade.dm

```
@@ -48,11 +48,12 @@
 	OnDeathInLife()
 
 /mob/living/simple_animal/shade/proc/OnDeathInLife()
-	if(stat == 2)
-		new /obj/item/ectoplasm (src.loc)
-		for(var/mob/M in viewers(src, null))
-			if((M.client && !( M.blinded )))
-				M.show_message("<span class='warning'>[src] lets out a contented sigh as their form unwinds.</span>")
-				ghostize()
-		qdel(src)
+	if(stat != DEAD)
 		return
+	var/obj/item/ectoplasm/dropped_ectoplasm = new
```

Why not just `new(loc)`?

---

That means that if drop_into fails or whatever, it'd wind up in `loc` instead of nullspace.